### ISSUE TITLE
Using RelatedFilter with name skipping a relationship doesn't seem to work

### ISSUE THREAD

**Issue by drewbrew** [philipn/django-rest-framework-filters#222]
TLDR: when using a `RelatedFilter` with `name='other_model__field'`, the related filter's attributes don't appear to be honored.

Example test case at https://github.com/drewbrew/related-filter-test.

## models.py

```python
from django.db import models


class Author(models.Model):
    name = models.CharField(max_length=200)


class Book(models.Model):
    title = models.CharField(max_length=50)
    author = models.ForeignKey(Author, models.DO_NOTHING, related_name='books')

    class Meta:
        unique_together = (('title', 'author'), )


class Chapter(models.Model):
    title = models.CharField(max_length=50)
    chapter_number = models.PositiveSmallIntegerField()
    book = models.ForeignKey(Book, models.DO_NOTHING, related_name='chapters')
    text = models.TextField()

    class Meta:
        unique_together = (('book', 'chapter_number'), )
```

## filters.py

```python
from rest_framework_filters import RelatedFilter
from rest_framework_filters.filterset import FilterSet

from relatedtest import models


class AuthorFilter(FilterSet):
    class Meta:
        model = models.Author
        fields = ['id', 'name']


class ChapterFilter(FilterSet):
    author = RelatedFilter(
        AuthorFilter,
        name='book__author',
        queryset=models.Author.objects.all(),
    )

    class Meta:
        model = models.Chapter
        fields = ['id', 'title', 'chapter_number', 'author']
```

# Expected behavior

I should be able to filter chapters by `author__name__icontains=smith` and get only chapters by authors who have `smith` in their name (case-insensitive) (i.e. `manage.py test` succeeds)

# Actual behavior

I get all chapters instead (i.e. `manage.py test` fails on `test_filtered_list()`)

# Environment
- Ubuntu 17.10 artful and 16.04 xenial, both amd64
- Tested on PostgreSQL and SQLite
- Django 2.0.2
- django-filter 1.1.0
- djangorestframework 3.7.7
- djangorestframework-filters 0.10.2
- pytz 2018.3

Any help would be greatly appreciated!

This package has been a seriously big help, so I also wanted to say thank you for your work.

**Comment by drewbrew** [comment]
FWIW, the problem still exists with current master (a8fb84ba2c71b5881637157990202d8db72c428b) and django-filter 2.0.0.dev1 if I change `name` to `field_name` per the D-F 2.0 migration guide.

**Comment by rpkilby** [comment]
Hi @drewbrew. The issue is that the `icontains` lookup hasn't been exposed on the author filter. Right now, the only valid filters are on the `name` and `id`. You would need to use the dictionary syntax of Meta.fields to expose lookups.

```python
class AuthorFilter(FilterSet):
    class Meta:
        model = models.Author
        fields = {
            'id': ['exact'],
            'name': ['exact', 'contains', 'icontains', ...],
        }
```

**Comment by drewbrew** [comment]
Wow, I can't believe I messed that one up. Thanks again!
